src/modules/vtol_att_control/vtol_type.cpp

```diff
@@ -127,25 +187,47 @@ void VtolType::update_mc_state()
 	_mc_pitch_weight = 1.0f;
 	_mc_yaw_weight = 1.0f;
 
+
 	// VTOL weathervane
-	_v_att_sp->disable_mc_yaw_control = false;
+	// TODO: Add check regarding data integrity. To avoid "pitch up"-bug at start.
+	if (_v_control_mode->flag_control_manual_enabled) {
+
+		if (_params->wv_manual && _v_control_mode->flag_control_velocity_enabled) {
+			wv_do_strategy();
+
+		}
 
-	if (_attc->get_pos_sp_triplet()->current.valid &&
-	    !_v_control_mode->flag_control_manual_enabled) {
+	} else if (_attc->get_pos_sp_triplet()->current.valid) {
 
-		if (_params->wv_takeoff && _attc->get_pos_sp_triplet()->current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF) {
-			_v_att_sp->disable_mc_yaw_control = true;
+		if (_params->wv_auto) {
+			wv_do_strategy();
+
+		} else if (_params->wv_takeoff
+			   && _attc->get_pos_sp_triplet()->current.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF) {
+			wv_do_strategy();
 
 		} else if (_params->wv_loiter
 			   && _attc->get_pos_sp_triplet()->current.type == position_setpoint_s::SETPOINT_TYPE_LOITER) {
-			_v_att_sp->disable_mc_yaw_control = true;
+			wv_do_strategy();
 
 		} else if (_params->wv_land && _attc->get_pos_sp_triplet()->current.type == position_setpoint_s::SETPOINT_TYPE_LAND) {
-			_v_att_sp->disable_mc_yaw_control = true;
+			wv_do_strategy();
 		}
 	}
 }
 
+void VtolType::wv_do_strategy()
+{
+	_v_att_sp->disable_mc_yaw_control = false;
+
+	if (_params->wv_strategy) {
+		_v_att_sp->disable_mc_yaw_control = true;
+
+	} else {
+		set_weather_vane_yaw_rate();
```

Comment:
@ivodre I don't quite understand this. If the the strategy is set to passive (value 0) then it will call set_weather_vane_yaw_rate(). Shouldn't it be the other way round?

Reply:
@RomanBapst we were just thinking about "enumerating" the strategies: 0 is ours, 1 the conventional one. But I see your point, will change it.